### PR TITLE
fix receipt bytes and path encoding

### PR DESCRIPTION
**Pull request by arthcp**

### CODE REVIEW

File: src/libs/ProofsUtil.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export default class ProofsUtil {
           blockHash: ethUtils.toBuffer(receipt.blockHash),
           parentNodes: stack.map(s => s.raw),
           root: ProofsUtil.getRawHeader(block).receiptTrie,
-          path: rlp.encode(receipt.transactionIndex),
+          path: Buffer.concat([Buffer.from('00', 'hex'), rlp.encode(receipt.transactionIndex)]),
```
[the remark this answers is not in the document]
Comment (arthcp) [reply src/libs/ProofsUtil.ts:153]: This code is common for pos and plasma flow in matic.js so separate changes are not needed for plasma.

**Review by jdkanani — APPROVED** [review]

File: src/libs/ProofsUtil.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ptsayli) [reply src/libs/ProofsUtil.ts:153]: this change will fail verification on plasma contracts, either we need to make a change in plasma contracts,
or handle this change for only pos

File: src/root/POSRootChainManager.ts [hunk 2]
```diff
@@ -30,13 +28,13 @@ export default class POSRootChainManager extends ContractsBase {
   constructor(options: MaticClientInitializationOptions, rootChain: RootChain, web3Client: Web3Client) {
     super(web3Client, options.network)
     this.posRootChainManager = new this.web3Client.parentWeb3.eth.Contract(
-      RootChainManagerArtifact.abi,
-      options.posRootChainManager
+      options.network.abi('RootChainManager', 'pos'),
```
Comment (ptsayli) [inline src/root/POSRootChainManager.ts:31]: nice, you've set default plasma

**Review by ptsayli — CHANGES_REQUESTED** [review]

File: src/libs/ProofsUtil.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (arthcp) [reply src/libs/ProofsUtil.ts:153]: Existing proof generation is flawed, it fails when there are multiple transactions in the burn transaction block for both pos and plasma. This fixes the issue for both.

**Review by ptsayli — APPROVED** [review]

**Pull request merged**